pmd-cli/src/test/java/net/sourceforge/pmd/cli/PmdCliTest.java

```diff
@@ -525,6 +525,25 @@ void minimumPriorityOptionNumeric() throws Exception {
         runCliSuccessfully("-d", srcDir.toString(), "-f", "text", "-R", RULESET_WITH_VIOLATION, "--minimum-priority", "2");
     }
 
+    @Test
+    void defaultThreadCount() throws Exception {
+        CliExecutionResult result = runCliSuccessfully("--debug", "--dir", srcDir.toString(), "--rulesets", RULESET_NO_VIOLATIONS);
+        result.checkStdErr(containsString("[DEBUG] Using " + Runtime.getRuntime().availableProcessors() + " threads for analysis"));
+    }
+
+    @Test
+    void monoThreadCount() throws Exception {
+        CliExecutionResult result = runCliSuccessfully("--debug", "--threads", "0", "--dir", srcDir.toString(), "--rulesets", RULESET_NO_VIOLATIONS);
+        result.checkStdErr(containsString("[DEBUG] Using single thread for analysis"));
```

Comment:
Just a suggestion, to make it significantly different from "using 1 thread" 

```suggestion
        result.checkStdErr(containsString("[DEBUG] Using main thread for analysis"));
```